src/server/pfs/server/api_server.go

```diff
@@ -163,7 +169,11 @@ func (a *apiServer) InspectRepo(ctx context.Context, request *pfs.InspectRepoReq
 
 // ListRepo implements the protobuf pfs.ListRepo RPC
 func (a *apiServer) ListRepo(request *pfs.ListRepoRequest, srv pfs.API_ListRepoServer) (retErr error) {
-	return a.driver.listRepo(srv.Context(), true /* includeAuth */, request.Type, request.Projects, srv.Send)
+	return errors.Wrap(dbutil.WithTx(srv.Context(), a.env.DB, func(ctx context.Context, tx *pachsql.Tx) error {
+		return a.driver.txnEnv.WithWriteContext(ctx, func(txnCxt *txncontext.TransactionContext) error {
+			return a.driver.listRepoInTransaction(srv.Context(), txnCxt, true, request.Type, request.Projects, srv.Send)
```

Comment:
This is the situation we were trying to avoid with the iterator.  We are calling `srv.Send` inside a transaction, which means that the gRPC client can keep open the transaction by reading slowly.

We are also performing a side effect inside the callback of dbutil.WithTx.   That means we could potentially be returning the results multiple times.  The `WithReadOnly` below might help with that, I'm not quite sure.

For streaming out over gRPC there are two options.
- Have the iterator take a `sqlx.DB` and do a transaction for a page at a time, read out the results.
- Read everything into a slice in memory in a transaction.  Wait for the transaction to succeed, then emit the results over gRPC.  This approach would not be appropriate for commits, but there are typically few enough repos that buffering in memory wouldn't be an issue.

Reply:
Okay, I think I'm going to implement a non-transactional list to handle this.

Reply:
This should be incorporated now